src/coreclr/pal/src/exception/signal.cpp

```diff
@@ -845,6 +845,15 @@ PAL_ERROR InjectActivationInternal(CorUnix::CPalThread* pThread)
     // We can get EAGAIN when printing stack overflow stack trace and when other threads hit
     // stack overflow too. Those are held in the sigsegv_handler with blocked signals until
     // the process exits.
+
+#ifdef __APPLE__
```

Comment:
Why guard this behind an ifdef? In what condition would we not want to return `ERROR_NOT_SUPPORTED`?